File: 2-Regresion/1-Regresion_Lineal_Simple/simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Tue Oct 11 19:43:24 2022

@author: Muerto
"""
# Regresion lineal simple

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importando dataset

dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:,:-1].values # tomamos todas las columnas, menos la ultima
Y = dataset.iloc[:,1].values# tomamos la ultima columna

# Dividir el dataset en conjunto de entrenamiento y conjunto de testing

from sklearn.model_selection import train_test_split

X_train, X_test, Y_train, Y_test = train_test_split(X,Y,test_size=1/3,random_state=0)

# Crear modelo de Regresion Lineal Simple con el conjunto de entrenamiento
from sklearn.linear_model import LinearRegression
regression = LinearRegression()

regression.fit(X_train ,Y_train)
# Predecir el cunjunto de test
Y_pred = regression.predict(X_test)

# Visualizar los resultados de entrenamiento
plt.scatter(X_train,Y_train, color="red")
plt.plot(X_train,regression.predict(X_train),color="blue")
plt.title("sueldo vs Años de experiencia")
plt.ylabel("Sueldo ($)")
plt.xlabel("Anos de experiencia")
plt.show()

# Visualizar los resultados de test
plt.scatter(X_test,Y_test, color="red")
plt.plot(X_train,regression.predict(X_train),color="blue")
plt.title("sueldo vs Años de experiencia")
plt.ylabel("Sueldo ($)")
plt.xlabel("Anos de experiencia")
plt.show()

File: 2-Regresion/1-Regresion_Lineal_Simple/simple_linear_regression.R
# Regresion Lineal Simple

# ----------------importando dataset----------------

dataset = read.csv('Salary_Data.csv')

# ----------------Dividiendo dataset en conjuntos----------------
# ----------------de entrenamiento y conjunto de testings----------------

# install.packages("caTools") # solo se necesita ejecutar una vez
# library(caTools)

# configurando semilla aleatoria para la division de datos
set.seed(0)
# se elige el porcentaje de los datos para el training en %
# se selecciona la columna de la variable dependiente o a predecir
split = sample.split(dataset$Salary,SplitRatio = 2/3)
print(split)
# Dividiendo el conjunto , True para el training
training_set = subset(dataset,split == TRUE)
# Dividiendo el conjunto , False para el test
testing_set = subset(dataset,split == FALSE)

# --------------------Crear modelo de Regresion Lineal--------------------
# --------------------Simple con el conjunto de entrenamiento--------------------
# visualisar documentacion de lm()
regressor=lm(formula=Salary ~ YearsExperience,data=training_set)
summary(regressor)

# --------------------Predecir el conjunto de test--------------------
# (modelo de prediccion,datos de testing)
y_pred=predict(regressor,newdata=testing_set)
print(y_pred)

# --------------------Visualizar los resultados de entrenamiento--------------------
# Para mostrar la grafica de datos de entrenamiento y la recta del modelo lineal

# Agregando componentes a mostrar
ggplot()+
  # Dobujando los puntos de entrenamiento
  geom_point(aes(x=training_set$YearsExperience,y=training_set$Salary),
             colour="red")+
  # Dobujando la linea de la prediccion, en base al entrenamiento
  geom_line(aes(x=training_set$YearsExperience,y=predict(regressor,newdata=training_set)),
            colour="blue")+
  ggtitle("Sueldo vs Años de experiencia(Entrenamiento)")+
  xlab("Sueldo ($)")+
  ylab("Anos de experiencia")

# --------------------Visualizar los resultados de test--------------------
# Para mostrar la grafica de datos de testing y la recta del modelo lineal


# Agregando componentes a mostrar
ggplot()+
  # Dobujando los puntos de testing
  geom_point(aes(x=testing_set$YearsExperience,y=testing_set$Salary),
             colour="red")+
  # Dobujando la linea de la prediccion, en base al testing
  geom_line(aes(x=testing_set$YearsExperience,y=y_pred),
            colour="blue")+
  ggtitle("Sueldo vs Años de experiencia(testing)")+
  xlab("Sueldo ($)")+
  ylab("Anos de experiencia")



